pygments/lexers/q.py

```diff
@@ -0,0 +1,235 @@
+"""
+    pygments.lexers.q
+    ~~~~~~~~~~~~~~~~~
+
+    Lexer for the Q programming language.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, words, include, bygroups, inherit
+from pygments.token import (
+    Comment,
+    Keyword,
+    Name,
+    Number,
+    Operator,
+    Punctuation,
+    String,
+    Text,
+    Whitespace,
+    Literal,
+    Generic,
+)
+
+__all__ = ["KLexer", "QLexer"]
+
+
+class KLexer(RegexLexer):
+    """
+    For `K <https://code.kx.com/>`_ source code.
+
+    .. versionadded:: 2.12
+    """
+
+    name = "K"
+    aliases = ["k"]
+    filenames = ["*.k"]
+
+    tokens = {
+        "whitespace": [
+            # hashbang script
+            (r"^#!.*", Comment.Hashbang),
+            # Comments
+            (r"^/\s*\n", Comment.Multiline, "comments"),
+            (r"(?<!\S)/.*", Comment.Single),
+            # Whitespace
+            (r"\s+", Whitespace),
+            # Strings
+            (r"\"", String.Double, "strings"),
+        ],
+        "root": [
+            include("whitespace"),
+            include("keywords"),
+            include("declarations"),
+        ],
+        "keywords": [
+            (
+                words(
+                    ("abs", "acos", "asin", "atan", "avg", "bin",
+                     "binr", "by", "cor", "cos", "cov", "dev",
+                     "delete", "div", "do", "enlist", "exec", "exit",
+                     "exp", "from", "getenv", "hopen", "if", "in",
+                     "insert", "last", "like", "log", "max", "min",
+                     "prd", "select", "setenv", "sin", "sqrt", "ss",
+                     "sum", "tan", "update", "var", "wavg", "while",
+                     "within", "wsum", "xexp"),
+                    suffix=r"\b",
+                ),
+                Operator.Word,
+            ),
+        ],
+        "declarations": [
+            # Timing
+            (r"^\\ts?", Comment.Preproc),
+            (
+                r"^(\\\w\s+[^/\n]*?)([/].*)",
+                bygroups(Comment.Preproc, Comment.Single),
+            ),
+            # Generic System Commands
+            (r"^\\\w+.*$", Comment.Preproc),
```

Comment:
While you are putting the final touch, please remove the useless `+` and `$` here.

Reply:
i've cleaned this up as well